pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -103,21 +98,36 @@ func (r *OperatorStepResolver) ResolveSteps(namespace string, _ SourceQuerier) (
 	updatedSubs := []*v1alpha1.Subscription{}
 	bundleLookups := []v1alpha1.BundleLookup{}
 	for name, op := range operators {
-		// TODO: added "is default channel" to sourceinfo out
-		// of convenience, which breaks map key equality here
-		// because it requires information that can't be
-		// gleaned from subscriptions alone. need to revisit
-		// TODO: this check also doesn't properly account for
-		// subscriptions made without a catalog specified,
-		// which means the sourceinfo won't match for the
-		// realized operator
+		// Find an existing subscription that resolves to this operator.
+		var (
+			existingSubscription *v1alpha1.Subscription
+			alreadyExists        bool
+		)
 		sourceInfo := *op.SourceInfo()
-		sourceInfo.DefaultChannel = false
-		_, isAdded := add[sourceInfo]
-		existingSubscription, subExists := subMap[sourceInfo]
+		for _, sub := range subs {
+			if sub.Spec.Package != sourceInfo.Package {
+				continue
+			}
+			if sub.Spec.Channel != "" && sub.Spec.Channel != sourceInfo.Channel {
```

Comment:
is `sub.Spec.Channel != ""` helpful here, if we only care about if the subs are the same? If they are both blank then we know both are pointing to the default channel.

Unless you're accounting for the case where the default channel changed in-between subscriptions, which makes sense.

Reply:
If the subscription doesn't specify a channel, it could be resolved from any channel (although the default channel is preferred). In that case, the operator's source would be the resolved channel.